extensions/metrics/src/interceptors/metrics.interceptor.ts

```diff
@@ -74,22 +86,41 @@ export class MetricsInterceptor implements Provider<Interceptor> {
     next: () => ValueOrPromise<T>,
   ) {
     MetricsInterceptor.setup();
-    const endGauge = MetricsInterceptor.gauge.startTimer({
-      targetName: invocationCtx.targetName,
-    });
-    const endHistogram = MetricsInterceptor.histogram.startTimer({
-      targetName: invocationCtx.targetName,
-    });
-    const endSummary = MetricsInterceptor.summary.startTimer({
+    const {request, response} = invocationCtx.parent as RequestContext;
+    const labelValues: LabelValues<LabelNames> = {
       targetName: invocationCtx.targetName,
-    });
+      method: request.method,
+      path: request.path,
+    };
+    const endGauge = MetricsInterceptor.gauge.startTimer();
+    const endHistogram = MetricsInterceptor.histogram.startTimer();
+    const endSummary = MetricsInterceptor.summary.startTimer();
     try {
-      MetricsInterceptor.counter.inc();
-      return await next();
+      const result = await next();
+      labelValues.statusCode = getStatusCodeFromResponse(response, result);
+      return result;
+    } catch (err) {
+      labelValues.statusCode = getStatusCodeFromError(err);
+      throw err;
     } finally {
-      endGauge();
-      endHistogram();
-      endSummary();
+      MetricsInterceptor.counter.inc(labelValues);
+      endGauge(labelValues);
+      endHistogram(labelValues);
+      endSummary(labelValues);
     }
   }
 }
+
+function getStatusCodeFromResponse(response: Response, result?: unknown) {
+  // interceptors are invoked before result is written to response,
+  // the status code for 200 responses without a result should be 204
+  const noContent = response.statusCode === 200 && result === undefined && 204;
+  return noContent || response.statusCode;
```

Comment:
`return noContent ?? response.statusCode;`

Reply:
`noContent` is of type `false | 204` which means if `||` is changed to `??` it would change the behavior

Reply:
I could also change the function implementation to 

```ts
function getStatusCodeFromResponse(response: Response, result?: unknown) {
  const noContent = response.statusCode === 200 && result === undefined;
  return noContent ? 204 : response.statusCode;
}
```

but I am no sure if that is better